### PR TITLE
Save recent search

### PR DESCRIPTION
**Pull request by morgandruesne**
## What does this PR do ?
This PR adds the possibility to have a query history and to be able to add queries as favorites and give them a name.

### How should this be manually tested?
  - Step 1 : do a search
  - Step 2 : go to the history page
  - Step 3 :  remane, add to favori or re-use a filter

### CODE REVIEW

File: src/components/Security/Users/List.vue [hunk 1]
```diff
@@ -238,14 +240,21 @@ export default {
...
-    onFiltersUpdated(newFilters) {
+    onFiltersUpdated(newFilters, isLoadHistory) {
       try {
         filterManager.save(
           newFilters,
           this.$router,
           this.index,
           this.collection
         )
+        if (!isLoadHistory) {
```
Comment (xbill82) [inline src/components/Security/Users/List.vue:251]:
```suggestion
        if (!fromHistory) {
```

File: src/services/filterManager.ts [hunk 2]
```diff
@@ -128,6 +129,94 @@ export const saveToLocalStorage = (filter, index, collection) => {
...
+    }
+  }
+  filters.push(filter)
+  localStorage.setItem(
+    `${HISTORY_LOCALSTORAGE_PREFIX}:${index}/${collection}`,
+    JSON.stringify(filters)
+  )
+}
+
+export const loadHistoyToLocalStorage = (index, collection) => {
```
Comment (xbill82) [inline src/services/filterManager.ts:204]:
```suggestion
export const loadHistoyFromLocalStorage = (index, collection) => {
```

File: src/services/filterManager.ts [hunk 3]
```diff
@@ -128,6 +129,94 @@ export const saveToLocalStorage = (filter, index, collection) => {
...
+}
+
+export const modifieHistoryRawFromLocalStorage = (
+  id,
+  newName,
+  bookmark,
+  index,
+  collection
+) => {
+  let filters = loadHistoyToLocalStorage(index, collection)
```
Comment (xbill82) [inline src/services/filterManager.ts:153]:
```suggestion
  let filters = loadHistoyFromLocalStorage(index, collection)
```

File: src/services/filterManager.ts [hunk 4]
```diff
@@ -128,6 +129,94 @@ export const saveToLocalStorage = (filter, index, collection) => {
   )
 }
 
+export const removeHistoryRawFromLocalStorage = (id, index, collection) => {
```
Comment (xbill82) [inline src/services/filterManager.ts:132]: If you use the word `History` you state that the function removes the whole history from localStorage. If it just removes one filter, the name should be
```suggestion
export const removeFilterFromLocalStorage = (id, index, collection) => {
```

File: src/services/filterManager.ts [hunk 5]
```diff
@@ -128,6 +129,94 @@ export const saveToLocalStorage = (filter, index, collection) => {
...
+  let filters = loadHistoyToLocalStorage(index, collection)
```
Comment (xbill82) [inline src/services/filterManager.ts:133]:
```suggestion
  let filters = loadHistoyFromLocalStorage(index, collection)
```

File: src/services/filterManager.ts [hunk 6]
```diff
@@ -128,6 +129,94 @@ export const saveToLocalStorage = (filter, index, collection) => {
...
+    })
+    .indexOf(id)
+  filters.splice(idIndex, 1)
+  localStorage.setItem(
+    `${HISTORY_LOCALSTORAGE_PREFIX}:${index}/${collection}`,
+    JSON.stringify(filters)
+  )
+}
+
+export const modifieHistoryRawFromLocalStorage = (
```
Comment (xbill82) [inline src/services/filterManager.ts:146]:
```suggestion
export const updateFilterInLocalStorage = (
```

File: src/services/filterManager.ts [hunk 7]
```diff
@@ -128,6 +129,94 @@ export const saveToLocalStorage = (filter, index, collection) => {
...
+    }
+    return filter
+  })
+  localStorage.setItem(
+    `${HISTORY_LOCALSTORAGE_PREFIX}:${index}/${collection}`,
+    JSON.stringify(filtersModified)
+  )
+}
+
+export const saveHistoyToLocalStorage = (filter, index, collection) => {
```
Comment (xbill82) [inline src/services/filterManager.ts:169]: Same as above: the naming suggests that the whole history is saved to localStorage (which, to me, seems the good thing to do), but if you just want to save one filter, you should call it 
```suggestion
export const saveFilterToLocalStorage = (filter, index, collection) => {
```

File: src/components/Common/Filters/FavorisFilter.vue [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    />
+  </ul>
+</template>
+
+<script>
+import * as filterManager from '../../../services/filterManager'
+import FavoriFilterIteam from './FavoriFilterIteam'
+
+export default {
+  name: 'FavorisFilter',
```
Comment (xbill82) [inline src/components/Common/Filters/FavorisFilter.vue:19]:
```suggestion
  name: 'FavoriteFilters',
```

(apply this to the file name)

File: src/components/Common/Filters/FavorisFilter.vue [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    index: String,
+    collection: String
+  },
+  data() {
+    return {
+      favoris: []
+    }
+  },
+  mounted() {
+    this.favoris = filterManager.LoadFavorisLocalStorage(
```
Comment (xbill82) [inline src/components/Common/Filters/FavorisFilter.vue]:
```suggestion
    this.favoris = filterManager.loadFavoritesFromLocalStorage(
```

File: src/components/Common/Filters/FavorisFilter.vue [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+      this.index,
+      this.collection
+    )
+  },
+  methods: {
+    onFavoriDelete(id) {
```
Comment (xbill82) [inline src/components/Common/Filters/FavorisFilter.vue:39]:
```suggestion
    onFavoriteDelete(id) {
```

File: src/components/Common/Filters/FavorisFilter.vue [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+          return favori.id
+        })
+        .indexOf(id)
+      this.favoris.splice(idIndex, 1)
+    }
+  },
+  watch: {
+    favoris: {
+      handler() {
+        filterManager.SaveFavorisLocalStorage(this.favoris, this.index, this.collection)
```
Comment (xbill82) [inline src/components/Common/Filters/FavorisFilter.vue]:
```suggestion
        filterManager.saveFavoriteToLocalStorage(this.favoris, this.index, this.collection)
```

File: src/components/Common/Filters/Filters.vue [hunk 12]
```diff
@@ -48,6 +48,12 @@
             @click="complexFiltersSelectedTab = 'history'"
             >Filter History</b-nav-item
           >
+          <b-nav-item
+            data-cy="Filters-favorisTab"
```
Comment (xbill82) [inline src/components/Common/Filters/Filters.vue:52]:
```suggestion
            data-cy="Filters-favoriteTab"
```

File: src/components/Common/Filters/Filters.vue [hunk 13]
```diff
@@ -48,6 +48,12 @@
...
+            :active="complexFiltersSelectedTab === 'favoris'"
```
Comment (xbill82) [inline src/components/Common/Filters/Filters.vue:53]:
```suggestion
            :active="complexFiltersSelectedTab === 'favorite'"
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 14]
```diff
@@ -1,51 +1,77 @@
 <template>
-  <div>
-    <HistoryFilterRaw
+  <ul class="list-group">
+    <HistoryFilterIteam
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue:3]:
```suggestion
    <HistoryFilterItem
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 15]
```diff
@@ -1,51 +1,77 @@
...
     />
-  </div>
+  </ul>
 </template>
 
 <script>
-import JsonFormatter from '../../../directives/json-formatter.directive'
 import * as filterManager from '../../../services/filterManager'
-import HistoryFilterRaw from './HistoryFilterRaw'
+import HistoryFilterIteam from './HistoryFilterIteam'
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue:17]:
```suggestion
import HistoryFilterItem from './HistoryFilterItem'
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 16]
```diff
@@ -1,51 +1,77 @@
...
 
 export default {
   name: 'HistoryFilter',
-  directives: {
-    JsonFormatter
-  },
   components: {
-    HistoryFilterRaw
+    HistoryFilterIteam
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue:22]:
```suggestion
    HistoryFilterItem
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 17]
```diff
@@ -1,51 +1,77 @@
...
   },
   props: {
     index: String,
     collection: String
   },
   data() {
     return {
-      filters: []
+      filters: [],
+      favoris: []
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue:31]:
```suggestion
      favorite: []
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 18]
```diff
@@ -1,51 +1,77 @@
...
     }
   },
   mounted() {
-    this.filters = filterManager.loadHistoyToLocalStorage(
+    this.filters = filterManager.LoadHistoyLocalStorage(
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue]:
```suggestion
    this.filters = filterManager.loadHistoyFromLocalStorage(
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 19]
```diff
@@ -1,51 +1,77 @@
...
       this.index,
       this.collection
     )
+    this.favoris = filterManager.LoadFavorisLocalStorage(
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue]:
```suggestion
    this.favoris = filterManager.loadFavoriteFromLocalStorage(
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 20]
```diff
@@ -1,51 +1,77 @@
...
+          return filter.id
+        })
+        .indexOf(id)
+      this.filters.splice(idIndex, 1)
+    }
+  },
+  watch: {
+    filters: {
+      handler() {
+        filterManager.SaveHistoyLocalStorage(
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue]:
```suggestion
        filterManager.saveHistoyToLocalStorage(
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 21]
```diff
@@ -1,51 +1,77 @@
...
+          this.filters,
+          this.index,
+          this.collection
+        )
+      },
+      deep: true
+    },
+    favoris: {
+      handler() {
+        filterManager.SaveFavorisLocalStorage(
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue]:
```suggestion
        filterManager.saveFavoriteToLocalStorage(
```

File: src/components/Common/Filters/HistoryFilterIteam.vue [hunk 22]
```diff
@@ -22,11 +23,11 @@
           </b-button>
           <b-button
             variant="link"
-            :title="'Bookmark Filter'"
-            @click="switchBookmark"
+            :title="'Favori Filter'"
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilterIteam.vue:26]:
```suggestion
            :title="'Favorite Filters'"
```

File: src/components/Common/Filters/HistoryFilterIteam.vue [hunk 23]
```diff
@@ -22,11 +23,11 @@
...
+            @click="favorisUpdate"
           >
             <i
-              :class="filter.bookmark === true ? 'fa fa-star' : 'far fa-star'"
+              :class="IsFavori() === true ? 'fa fa-star' : 'far fa-star'"
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilterIteam.vue:30]:
```suggestion
              :class="IsFavorite() === true ? 'fa fa-star' : 'far fa-star'"
```

File: src/components/Common/Filters/HistoryFilterIteam.vue [hunk 24]
```diff
@@ -22,11 +23,11 @@
           </b-button>
           <b-button
             variant="link"
-            :title="'Bookmark Filter'"
-            @click="switchBookmark"
+            :title="'Favori Filter'"
+            @click="favorisUpdate"
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilterIteam.vue:27]:
```suggestion
            @click="favoriteUpdate"
```

File: src/components/Common/Filters/HistoryFilterIteam.vue [hunk 25]
```diff
@@ -57,84 +58,73 @@
     <b-modal
       size="lg"
       :id="`changeNameHistoryFilter-${filter.id}`"
-      title="you want to change this filter ?"
+      title="you want to change this filter name ?"
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilterIteam.vue:61]:
```suggestion
      title="Do you want to change the name of this filter?"
```

**Comment by xbill82** [comment]
I like the UI and the behavior looks OK 👌 , nice job! Need to reword (almost all) the variables and methods (functions must not start with an uppercase letter) but the feature looks neat.
Only thing I would add: the empty-state of the filter history and the favorites (i.e. a nice text box saying "There's currently no filter in the history" and "you currently have no favorite filter")

File: src/components/Common/Filters/HistoryFilter.vue [hunk 26]
```diff
@@ -1,25 +1,25 @@
 <template>
   <ul class="list-group">
-    <HistoryFilterIteam
+    <HistoryFilterItem
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue:3]: Sorry...
```suggestion
    <FilterHistoryItem
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 27]
```diff
@@ -54,18 +54,18 @@ export default {
...
         )
       },
       deep: true
     },
-    favoris: {
+    favorite: {
       handler() {
-        filterManager.SaveFavorisToLocalStorage(
-          this.favoris,
+        filterManager.saveFavoriteToLocalStorage(
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue:67]: sorry...
```suggestion
        filterManager.saveFavoritesToLocalStorage(
```

File: src/components/Common/Filters/HistoryFilterItem.vue [hunk 28]
```diff
@@ -58,7 +58,7 @@
     <b-modal
       size="lg"
       :id="`changeNameHistoryFilter-${filter.id}`"
-      title="you want to change this filter name ?"
+      title="Do you want to change the name of this filter?"
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilterItem.vue:61]:
```suggestion
      title="Edit favorite filter name"
```

File: src/components/Common/Filters/HistoryFilterItem.vue [hunk 29]
```diff
@@ -73,12 +73,12 @@ import * as filterManager from '../../../services/filterManager'
 import { mapGetters } from 'vuex'
 
 export default {
-  name: 'HistoryFilterIteam',
+  name: 'HistoryFilterItem',
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilterItem.vue:76]:
```suggestion
  name: 'FilterHistoryItem',
```

File: src/components/Common/Filters/HistoryFilterItem.vue [hunk 30]
```diff
@@ -90,8 +90,8 @@ export default {
     ...mapGetters('kuzzle', ['wrapper'])
   },
   methods: {
-    IsFavori() {
-      let idIndex = this.favoris
+    IsFavorite() {
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilterItem.vue:93]:
```suggestion
    isFavorite() {
```

Comment (xbill82) [reply src/components/Common/Filters/HistoryFilterItem.vue:93]: also, should be a computed property

File: src/components/Common/Filters/HistoryFilterItem.vue [hunk 31]
```diff
@@ -105,22 +105,22 @@ export default {
...
           })
           .indexOf(this.filter.id)
-        this.favoris.splice(idIndex, 1)
+        this.favorite.splice(idIndex, 1)
       } else {
-        this.favoris.push(this.filter)
+        this.favorite.push(this.filter)
       }
     },
     SubmitChange() {
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilterItem.vue:120]:
```suggestion
    submitChange() {
```

File: src/components/Common/Filters/HistoryFilterItem.vue [hunk 32]
```diff
@@ -105,22 +105,22 @@ export default {
...
-      this.favorisUpdate()
+      this.favoriteUpdate()
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilterItem.vue:121]:
```suggestion
      this.updateFavorite()
```

File: src/components/Data/Documents/Page.vue [hunk 33]
```diff
@@ -471,7 +471,7 @@ export default {
           this.collection
         )
         if (!isLoadHistory) {
-          filterManager.NewHistoyToLocalStorage(
+          filterManager.newHistoyLocalStorage(
```
Comment (xbill82) [inline src/components/Data/Documents/Page.vue:474]: What is the purpose of this?

File: src/components/Common/Filters/FavoriteFilterItem.vue [hunk 34]
```diff
@@ -0,0 +1,113 @@
...
+    </b-modal>
+  </li>
+</template>
+
+<script>
+import * as filterManager from '../../../services/filterManager'
+import { mapGetters } from 'vuex'
+
+export default {
+  name: 'FavoriFilterIteam',
```
Comment (xbill82) [inline src/components/Common/Filters/FavoriteFilterItem.vue:65]:
```suggestion
  name: 'FavoriteFilterItem',
```

File: src/components/Common/Filters/FavoriteFilterItem.vue [hunk 35]
```diff
@@ -0,0 +1,113 @@
...
+  props: {
+    index: String,
+    collection: String,
+    favori: Object
+  },
+  data() {
+    return {
+      expanded: false,
+      oldName: this.favori.name,
```
Comment (xbill82) [inline src/components/Common/Filters/FavoriteFilterItem.vue:74]:
```suggestion
      oldName: this.favorite.name,
```

File: src/components/Common/Filters/FavoriteFilterItem.vue [hunk 36]
```diff
@@ -0,0 +1,113 @@
...
+  },
+  methods: {
+    cancelChange() {
+      this.favori.name = this.oldName
+    },
+    openModal() {
+      this.$bvModal.show('changeNameHistoryFilter-' + this.favori.id)
+    },
+    deleteFavori() {
+      this.$emit('favoris-delete', this.favori.id)
```
Comment (xbill82) [inline src/components/Common/Filters/FavoriteFilterItem.vue:88]:
```suggestion
      this.$emit('favorite-delete', this.favori.id)
```

File: src/components/Common/Filters/FavoriteFilters.vue [hunk 37]
```diff
@@ -0,0 +1,57 @@
...
+    index: String,
+    collection: String
+  },
+  data() {
+    return {
+      favoris: []
+    }
+  },
+  mounted() {
+    this.favoris = filterManager.loadFavoritesFromLocalStorage(
```
Comment (xbill82) [inline src/components/Common/Filters/FavoriteFilters.vue:33]:
```suggestion
    this.favorites = filterManager.loadFavoritesFromLocalStorage(
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 38]
```diff
@@ -0,0 +1,77 @@
...
+      :filter="filter"
+      :favorite="favorite"
+      @filters-delete="onFiltersDelete"
+    />
+  </ul>
+</template>
+
+<script>
+import * as filterManager from '../../../services/filterManager'
+import HistoryFilterItem from './HistoryFilterItem'
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue:17]:
```suggestion
import FilterHistoryItem from './HistoryFilterItem'
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 39]
```diff
@@ -0,0 +1,77 @@
...
+
+export default {
+  name: 'HistoryFilter',
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue:20]:
```suggestion
  name: 'FilterHistory',
```

File: src/components/Common/Filters/HistoryFilterItem.vue [hunk 40]
```diff
@@ -0,0 +1,152 @@
...
+      if (idIndex !== -1) {
+        return true
+      } else {
+        return false
+      }
+    },
+    cancelChange() {
+      this.filter.name = this.oldName
+    },
+    favoriteUpdate() {
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilterItem.vue:108]:
```suggestion
    updateFavorite() {
```

File: src/components/Common/Filters/FavoriteFilters.vue [hunk 41]
```diff
@@ -1,13 +1,23 @@
...
+        :key="i"
+        :index="index"
+        :collection="collection"
+        :favorite="favori"
+        @favoris-delete="onFavoriteDelete"
+      />
+    </template>
+    <template v-else>
+      <h4 class="text-secondary text-center">
+        You don't have any favorites filters.
```
Comment (xbill82) [inline src/components/Common/Filters/FavoriteFilters.vue:15]:
```suggestion
        You don't have any favorite filters.
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 42]
```diff
@@ -1,14 +1,24 @@
...
+        :index="index"
+        :collection="collection"
+        :filter="filter"
+        :favorite="favorite"
+        @filters-delete="onFiltersDelete"
+      />
+    </template>
+    <template v-else>
+      <h4 class="text-secondary text-center">
+        You don't have any history filters.
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue:16]:
```suggestion
        You haven't performed any search yet.
```

File: src/components/Common/Filters/HistoryFilter.vue [hunk 43]
```diff
@@ -1,14 +1,24 @@
...
+      </h4>
+      <p class="text-secondary text-center">
+        You can add more by making a query.
```
Comment (xbill82) [inline src/components/Common/Filters/HistoryFilter.vue:19]:
```suggestion
        You'll find all your searches in this history.
```

File: test/e2e/cypress/integration/single-backend/search.spec.js [hunk 44]
```diff
@@ -537,4 +537,185 @@ describe('Search', function() {
...
+    cy.get('[data-cy=QuickFilter-optionBtn]').click()
+    cy.get('[data-cy=Filters-basicTab]').click()
+    cy.get('[data-cy="BasicFilter-attributeSelect--0.0"]').select('lastName')
+    cy.get('[data-cy="BasicFilter-valueInput--0.0"]').type('4')
+    cy.get('[data-cy=BasicFilter-submitBtn]').click()
+    cy.get('[data-cy=QuickFilter-displayActiveFilters]').click()
+    cy.get('[data-cy=Filters-historyTab]').click()
+    cy.get('[data-cy="FilterHistoryItem--0"]')
+  })
+  it('should add mamy search in history', () => {
```
Comment (xbill82) [inline test/e2e/cypress/integration/single-backend/search.spec.js:570]: ❓

File: test/e2e/cypress/integration/single-backend/search.spec.js [hunk 45]
```diff
@@ -537,4 +537,185 @@ describe('Search', function() {
...
+    cy.get('[data-cy=Filters-historyTab]').click()
+    cy.get('[data-cy="FilterHistoryItem--0"]')
+    cy.get('[data-cy="FilterHistoryItem-Add-Favorite--0"]').click()
+    cy.get('[data-cy=Filters-favoriteTab]').click()
+    cy.get('[data-cy="FilterFavoriItem--0"]')
+    cy.get('[data-cy="FilterFavoriItem-Remove--0"]').click()
+    cy.get('[data-cy="FilterFavoriItem--0"]').should('not.exist')
+    cy.contains("You don't have any favorite filters.")
+  })
+  it('should by able to perform search from history', () => {
```
Comment (xbill82) [inline test/e2e/cypress/integration/single-backend/search.spec.js:681]:
```suggestion
  it('should be able to perform search from history', () => {
```

File: test/e2e/cypress/integration/single-backend/search.spec.js [hunk 46]
```diff
@@ -567,7 +567,7 @@ describe('Search', function () {
     cy.get('[data-cy=Filters-historyTab]').click()
     cy.get('[data-cy="FilterHistoryItem--0"]')
   })
-  it('should add mamy search in history', () => {
+  it('should be able to multi-searchable and must appear in the history section.', () => {
```
Comment (xbill82) [inline test/e2e/cypress/integration/single-backend/search.spec.js:570]: TIHS NOT IZ ANGLISCH

**Review by xbill82 — APPROVED** [review]

File: src/services/filterManager.ts [hunk 47]
```diff
@@ -128,6 +130,80 @@ export const saveToLocalStorage = (filter, index, collection) => {
...
+}
+
+export const saveHistoyToLocalStorage= (filters, index, collection) => {
+  localStorage.setItem(
+    `${HISTORY_LOCALSTORAGE_PREFIX}:${index}/${collection}`,
+    JSON.stringify(filters)
+  )
+}
+
+export const newHistoyLocalStorage = (filter, index, collection) => {
```
Comment (xbill82) [inline src/services/filterManager.ts:162]:
```suggestion
export const addNewHistoryItemAndSave = (filter, index, collection) => {
```

**Review by xbill82 — APPROVED** [review]

**Review by xbill82 — APPROVED** [review]

File: src/components/Common/Filters/FilterHistoryItem.vue [hunk 48]
```diff
@@ -0,0 +1,154 @@
...
+      expanded: false,
+      oldName: this.filter.name
+    }
+  },
+  computed: {
+    ...mapGetters('kuzzle', ['wrapper'])
+  },
+  methods: {
+    isFavorite() {
+      let idIndex = this.favorite
```
Comment (Njuelle) [inline src/components/Common/Filters/FilterHistoryItem.vue:96]: nitpicking
```suggestion
      const idIndex = this.favorite
```

File: src/components/Common/Filters/FilterHistoryItem.vue [hunk 49]
```diff
@@ -0,0 +1,154 @@
...
+      } else {
+        return false
+      }
+    },
+    cancelChange() {
+      this.filter.name = this.oldName
+    },
+    updateFavorite() {
+      if (this.isFavorite()) {
+        let idIndex = this.favorite
```
Comment (Njuelle) [inline src/components/Common/Filters/FilterHistoryItem.vue:112]: nitpicking
```suggestion
        const idIndex = this.favorite
```

File: src/components/Common/Filters/FilterHistoryItem.vue [hunk 50]
```diff
@@ -0,0 +1,154 @@
...
+        this.$parent.$emit(
+          'filter-basic-submitted',
+          this.filter.basic,
+          this.filter.sorting,
+          true
+        )
+      }
+    },
+    getFilter() {
+      let loadedFilter = Object.assign(new filterManager.Filter(), this.filter)
```
Comment (Njuelle) [inline src/components/Common/Filters/FilterHistoryItem.vue:145]: nitpicking
```suggestion
      const loadedFilter = Object.assign(new filterManager.Filter(), this.filter)
```

File: src/components/Common/Filters/Filters.vue [hunk 51]
```diff
@@ -233,22 +272,23 @@ export default {
         })
       )
     },
-    onBasicFilterUpdated(filter, sorting) {
+    onBasicFilterUpdated(filter, sorting, isLoadHistory) {
```
Comment (Njuelle) [inline src/components/Common/Filters/Filters.vue:275]: I don't understand what the variable `isLoadHistory` is for, help me to understand plz :)
Maybe a wording/naming problem?

File: src/components/Common/Filters/HistoryFilter.vue [hunk 52]
```diff
@@ -0,0 +1,89 @@
...
+      this.collection
+    )
+    this.favorite = filterManager.loadFavoritesFromLocalStorage(
+      this.index,
+      this.collection
+    )
+  },
+  methods: {
+    onFiltersDelete(id) {
+      let idIndex = this.filters
```
Comment (Njuelle) [inline src/components/Common/Filters/HistoryFilter.vue:58]: nitpicking
```suggestion
      const idIndex = this.filters
```

File: src/services/filterManager.ts [hunk 53]
```diff
@@ -128,6 +130,80 @@ export const saveToLocalStorage = (filter, index, collection) => {
...
+export const addNewHistoryItemAndSave = (filter, index, collection) => {
+  if (!index || !collection) {
+    throw new Error(
+      'Cannot save filters to localstorage if no index or collection are specified'
+    )
+  }
+  if (filter.active === null) {
+    return
+  }
+  let filters = loadHistoyFromLocalStorage(index, collection)
```
Comment (Njuelle) [inline src/services/filterManager.ts:171]: nitpicking
```suggestion
  const filters = loadHistoyFromLocalStorage(index, collection)
```

File: src/services/filterManager.ts [hunk 54]
```diff
@@ -128,6 +130,80 @@ export const saveToLocalStorage = (filter, index, collection) => {
...
+  let date = new Date(Date.now())
```
Comment (Njuelle) [inline src/services/filterManager.ts:172]: nitpicking
```suggestion
  const date = new Date(Date.now())
```

**Review by Njuelle — CHANGES_REQUESTED** [review]

File: src/components/Common/Filters/Filters.vue (hunk 51 above)
Comment (morgandruesne) [reply src/components/Common/Filters/Filters.vue:275]: this variable is used not to re-save a search in the history if we launches the search from this one

Comment (xbill82) [reply src/components/Common/Filters/Filters.vue:275]:
```suggestion
    onBasicFilterUpdated(filter, sorting, loadedFromHistory) {
```

**Review by Njuelle — APPROVED** [review]

**Pull request merged**